source/game.database.bmx

```diff
@@ -252,7 +252,7 @@ Type TDatabaseLoader
 		'load according to version
 		Select version
 '			case 2	LoadV2(xml)
-			Case 2	TLogger.Log("TDatabase.Load()", "CANNOT LOAD DB ~q" + fileURI + "~q (version "+version+") - version 2 is deprecated. Upgrade to V3 please." , LOG_LOADING)
+			Case 2	TLogger.Log("TDatabase.Load()", "CANNOT LOAD DB ~q" + fileURI + "~q (version "+version+") - version 2 (deprecated) or invalid XML. Use correct V3-format please." , LOG_LOADING)
```

Comment:
Ich wuerde da nochmal selbst nachschauen, wieso "version 2" rauskommt. Das Anpassen der Fehlermeldung sollte unnoetig sein - wenn invalides XML "korrekt" erkannt wuerde.

Reply:
Unabhaengig davon sind "v2"-Datenbanken seit ... vielen Jahren... nicht mehr im Einsatz. Ich denke der Support dafuer kann "raus" (und als "UNKNOWN VERSION" abgestempelt werden).

Reply:
Das sollte es tun:
```
		'recognize version (if versionNode is Null, defaultValue -1 is returned
		Local versionNode:TxmlNode = xml.FindRootChild("version")
		Local version:Int = xml.FindValueInt(versionNode, "value", -1)

		'load according to version
		Select version
			Case 3	LoadV3(xml)
			Default	TLogger.Log("TDatabase.Load()", "CANNOT LOAD DB ~q" + fileURI + "~q (version "+version+") - UNKNOWN VERSION or INVALID XML FILE." , LOG_LOADING)
		End Select
```
Die fehlerhafte Zeilennummer schau ich mir noch an.

Reply:
Fehlerhafte Zeilennummer: Der Fehler ist in der originalen MXML-Bibliothek - wurde dort schon gefixt, aber scheinbar nutzen wir in "text.mod/mxml.mod" noch nicht diese korrigierte Fassung.

Reply:
Bei mit kommt das Logging für die XML-Fehler aus main nicht. Da kann man ja einen Callback registrieren. Im obigen Code bekommt man ja nicht mit wo der Fehler ist. Die Version ist -1, da der Versionsknoten nicht gefunden wird, OK. Dann weiß man dass es invalides XML ist. Aber die Ausgabe, dass es ein Fehler ist, scheint nur ein print zu sein, der im "kompilierten" Spiel gar nicht im Log auftauchen würde. Die reparierte Zeilennummer hilft nicht, wenn der Callback nicht funktioniert.

Reply:
Bei mir wird das geloggt.

Ist aber am Ende auch egal, da ich lokal die Codezeilen schon umgeschrieben habe. Ich warte aber noch darauf, dass Brucey "mxml" aktualisiert (erspart das manuelle anpassen einer Zeile dort, damit die Zeilennummern fuer die "duplicate"-Fehler hinhauen). Dann kann ich einen PR stellen, der das Registrieren des "mxml-Fehler-Callbacks" um einen "Unset" erweitert.
(Erweiternd hierzu: Ich habe wohl lange genug "gedraengelt", da Brucey nun auto-Build-Scripts fuer BlitzMaxNG gebaut hat und nun "weekly" neue Versionen zusammengestellt werden)

Lokal melde fuege ich sozusagen vor dem XML-Laden einen Callback ein, der TLogger aufruft, und nach dem Laden wird der Callback wieder abgemeldet (damit die Fehlermeldung eine andere "Quelle" als "TDatabaseLoader" angeben kann).

Sprich der PR hier kann gern das "Abhaerten" enthalten, die Fehlermeldungsgeschichte koennen wir aber dann gern in einem anderen PR abhandeln.

Reply:
Hat sich erledigt, da Fehlermeldung selbst in anderem PR behandelt